pkg/services/publicdashboards/api/middleware.go

```diff
@@ -11,6 +14,31 @@ func SetPublicDashboardFlag() func(c *models.ReqContext) {
 	}
 }
 
+func RequiresValidAccessToken(publicDashboardService publicdashboards.Service) func(c *models.ReqContext) {
+	return func(c *models.ReqContext) {
+		accessToken, ok := web.Params(c.Req)[":accessToken"]
+
+		// Check access token is present on the request
+		if !ok || len(accessToken) < 1 {
+			c.JsonApiErr(401, "Unauthorized: access token not provided", nil)
```

Comment:
I think this should return a `Bad Request` error. It feels weird to return a 401 in a public endpoint.